modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -4094,6 +4096,135 @@ public void testPartialUpdateObjectEntryWithAttachmentObjectField()
 		objectDefinitionLocalService.deleteObjectDefinition(objectDefinition);
 	}
 
+	@Test
+	public void testPartialUpdateObjectEntryWithLocalizedAttachmentObjectField()
+		throws Exception {
+
+		ObjectDefinition objectDefinition = _createObjectDefinition(
+			Collections.singletonList(
+				new AttachmentObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap("localizedAttachment")
+				).localized(
+					true
+				).name(
+					"localizedAttachment"
+				).objectFieldSettings(
+					Arrays.asList(
+						new ObjectFieldSettingBuilder(
+						).name(
+							ObjectFieldSettingConstants.
+								NAME_ACCEPTED_FILE_EXTENSIONS
+						).value(
+							"txt"
+						).build(),
+						new ObjectFieldSettingBuilder(
+						).name(
+							ObjectFieldSettingConstants.NAME_FILE_SOURCE
+						).value(
+							ObjectFieldSettingConstants.VALUE_USER_COMPUTER
+						).build(),
+						new ObjectFieldSettingBuilder(
+						).name(
+							ObjectFieldSettingConstants.NAME_MAX_FILE_SIZE
+						).value(
+							"100"
+						).build())
+				).build()));
+
+		long tempFileEntryId1 = _addTempFileEntry(
+			objectDefinition.getPortletId(), StringUtil.randomString());
+		long tempFileEntryId2 = _addTempFileEntry(
+			objectDefinition.getPortletId(), StringUtil.randomString());
+
+		ObjectEntry objectEntry = _defaultObjectEntryManager.addObjectEntry(
+			_simpleDTOConverterContext, objectDefinition,
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						"localizedAttachment_i18n",
+						HashMapBuilder.put(
+							"en_US", tempFileEntryId1
+						).put(
+							"pt_BR", tempFileEntryId2
+						).build()
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		AssertUtils.assertFailure(
+			NoSuchFileEntryException.class,
+			StringBundler.concat(
+				"No FileEntry exists with the key {fileEntryId=",
+				tempFileEntryId1, "}"),
+			() -> _dlAppLocalService.getFileEntry(tempFileEntryId1));
+		AssertUtils.assertFailure(
+			NoSuchFileEntryException.class,
+			StringBundler.concat(
+				"No FileEntry exists with the key {fileEntryId=",
+				tempFileEntryId2, "}"),
+			() -> _dlAppLocalService.getFileEntry(tempFileEntryId2));
+
+		objectEntry = _defaultObjectEntryManager.partialUpdateObjectEntry(
+			_simpleDTOConverterContext, objectDefinition, objectEntry.getId(),
+			new ObjectEntry() {
+				{
+					properties = Collections.emptyMap();
+				}
+			});
+
+		Map<String, Serializable> localizedValues =
+			(Map<String, Serializable>)objectEntry.getPropertyValue(
+				"localizedAttachment_i18n");
+
+		FileEntry persistedFileEntry1 = (FileEntry)localizedValues.get("en_US");
```

Comment:
I would rename it to `fileEntry1`, but if you really want to give the impression that it persisted, I would rename it to `actualFileEntry1` (but maybe Brian would look for the variable `expectedFileEntry1`).

Reply:
All right, I renamed to `fileEntry1` and `fileEntry2`